src/sciline/pipeline.py

```diff
@@ -295,17 +296,18 @@ def _copy_node(
             )
         selected = {
             label: arg
-            for label, arg in zip(provider.labels, provider.arg_spec.args)
+            for label, arg in zip(provider.labels, provider.arg_spec.args, strict=True)
             if label in labels
         }
         return SeriesProvider(selected.keys(), provider.row_dim, args=selected.values())
 
     def _find_grouping_node(self, index_name: Key, subgraph: Graph) -> type:
-        ends: List[type] = []
-        for key in subgraph:
-            if get_origin(key) == Series and get_args(key)[0] == index_name:
-                # Because of the succeeded get_origin we know it is a type
-                ends.append(key)  # type: ignore[arg-type]
+        ends: List[type] = [
```

Comment:
Still `List` with capital 'L'. Did you run the upgrader for type hints?

Reply:
I did not. I thought we added pyupgrade to the ruff rules https://github.com/scipp/copier_template/blob/afbe5c758fbc9e42b4379939e833e1f746187dd7/template/pyproject.toml.jinja#L67 but I guess not?

Reply:
No, you would have to run it manually